File: tests/src/com.google.protobuf/protobuf-javalite/3.25.3/src/test/java/com_google_protobuf/protobuf_javalite/ByteBufferWriterTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.assertj.core.api.Assertions.assertThat;

import com.google.protobuf.ByteString;
import org.junit.jupiter.api.Test;

public class ByteBufferWriterTest {
    private static final String BYTE_BUFFER_WRITER_CLASS_NAME = "com.google.protobuf.ByteBufferWriter";

    @Test
    void classInitializationDiscoversFileOutputStreamChannelField() throws Exception {
        Class<?> writerClass = Class.forName(
                BYTE_BUFFER_WRITER_CLASS_NAME,
                true,
                ByteString.class.getClassLoader()
        );

        assertThat(writerClass.getName()).isEqualTo(BYTE_BUFFER_WRITER_CLASS_NAME);
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_2.13/1.1.5/src/test/scala/org_apache_pekko/pekko_protobuf_v3_2_13/ByteBufferWriterTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_2_13

import org.apache.pekko.protobufv3.internal.ByteString
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ByteBufferWriterTest {
  @Test
  def initializesByteBufferWriterRuntimeSupport(): Unit = {
    val className: String = "org.apache.pekko.protobufv3.internal.ByteBufferWriter"
    val writerClass: Class[_] = Class.forName(className, true, classOf[ByteString].getClassLoader)

    assertThat(writerClass.getName).isEqualTo(className)
  }
}
